superset/db_engine_specs/presto.py

```diff
@@ -392,28 +392,49 @@ def update_impersonation_config(
 
     @classmethod
     def get_table_names(
-        cls, database: Database, inspector: Inspector, schema: Optional[str]
+        cls,
+        database: Database,
+        inspector: Inspector,
+        schema: Optional[str],
     ) -> List[str]:
-        tables = super().get_table_names(database, inspector, schema)
-        if not is_feature_enabled("PRESTO_SPLIT_VIEWS_FROM_TABLES"):
-            return tables
+        """
+        Get all the physical table names from the specified schema.
+
+        Presto's perspective is the table concept represents both physical and non-
+        physical tables (views).
+
+        Note if the schema is omitted, per PyHive's implementation, it defaults to
+        returning all the tables within the database.
+
+        :param database: The database to inspect
+        :param inspector: The SQLAlchemy inspector
+        :param schema: The schema to inspect
+        :returns: The physical table names
+        """
 
-        views = set(cls.get_view_names(database, inspector, schema))
-        actual_tables = set(tables) - views
-        return list(actual_tables)
+        return list(
```

Comment:
Granted this logic is the same for both Presto and Trino and thus could go in the base class, however there's a subtle difference in how these engines handle an undefined schema which is documented.